src/test/basics/PerfLog_test.cpp

```diff
@@ -0,0 +1,1075 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2018 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+
+#include <ripple/basics/PerfLog.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/json/json_reader.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/rpc/impl/Handler.h>
+#include <test/jtx/Env.h>
+#include <atomic>
+#include <chrono>
+#include <random>
+#include <string>
+#include <thread>
+
+//------------------------------------------------------------------------------
+
+namespace ripple {
+
+class PerfLog_test : public beast::unit_test::suite
+{
+    enum class WithFile : bool
```

Comment:
I'm personally OK with this. However, bool enums like this have been verboten in the past. This is really Nik's call if he wants to losen the style. We've also been looser with style in test files, which is an argument to allow it.